docs_src/microservices/core/data/Ch-CoreData.md

```diff
@@ -1,102 +1,170 @@
-latency through this layer and storage needs at the network edge, but
-the cost is having no historical data to use for operations based on
-changes over time, and only minimal device actuation decisions, based on
-single event data, at the edge.
+By default, core data persists all data collected by devices sent to it. However, when the data is too sensitive to keep
+at the edge, or there is no use for the data at the edge by other local services (e.g. by an analytics micro service), the data
+can be "streamed" through core data without persisting it. A
+configuration change to core data (PersistData=false) has core data
+send data to the application services without
+persisting the data. This option has the advantage of reducing
+latency through this layer and storage needs at the network edge.  But
+the cost is having no historical data to use for analytics that need to look back in time to make a decision.
+
+## Events and Readings
+
+Data collected from sensors is marshalled into EdgeX event and reading objects (delivered as JSON objects in service REST calls to core data).  An event represents a collection of one or more sensor readings.  Some sensors or devices are only providing a single value – a single reading - at a time. Other sensors spew multiple values whenever they are read.
+
+An event must have at least one reading.  Events are associated to a sensor or device – the “thing” that sensed the environment and produced the readings.  Readings represent a sensing on the part of a device or sensor.  Readings only exist as part of (are owned by) an event.  Readings are essentially a simple key/value pair of what was sensed (the key) and the value sensed (the value).  A reading may include other bits of information to provide more context (for example data type information) for the users of that data.  Consumers of the reading data could include things like user interfaces, data visualization systems and analytics tools.
+
+In the diagram below, an example event/reading collection is depicted.  The event coming from the “motor123” device has two readings (or sensed values).  The first reading indicates that the motor123 device reported the pressure of the motor was 1300 (the unit of measure might be something like PSI).
+
+![image](EdgeX_Event-Reading.png)
+
+The value type property (shown as type above) on the reading lets the consumer of the information know that the value is an integer, base 64.  The second reading indicates that the motor123 device also reported the temperature of the motor was 120 at the same time it reported the pressure (perhaps in degrees Fahrenheit).
 
 ## Data Model
 
-The following diagram shows the Data Model for Core Data.
+The following diagram shows the Data Model for core data.  Device services send Event objects containing a collection or Readings to core data when a device captures a sensor reading.
 
 ![image](EdgeX_CoreDataModel.png)
```

Comment:
I'm probably in the minority here. But the swagger docs tell me this same content in a much more digestable way. I question whether or not this is needed. It feels a bit un-approachable and outdated to provide a UML diagram.

Reply:
I received a couple of requests to update the diagrams, so someone out there is using them.  I also find it convenient sometimes to pull up the diagram when presenting information about some of our models to others.   I did update all the diagrams for this documents refresh.  Admittedly they can become out of date, but I still think they hold some value.